snuba_sdk/timeseries.py

```diff
@@ -1,15 +1,32 @@
 from __future__ import annotations
 
-from dataclasses import dataclass, replace
-from typing import Any
+from dataclasses import dataclass, fields, replace
+from typing import Any, Sequence
 
 from snuba_sdk.column import Column
 from snuba_sdk.conditions import Condition
-from snuba_sdk.expressions import Expression, InvalidExpressionError, is_literal
+from snuba_sdk.expressions import InvalidExpressionError, is_literal, list_type
+from snuba_sdk.orderby import Direction
+
+
+class InvalidTimeseriesError(Exception):
+    pass
+
+
+VALID_ENTITIES = set(
```

Comment:
Some thoughts on entity validation. After this change, we would have snuba entity names defined in 3 places: sentry, snuba-sdk, and snuba itself. Entity resolution currently takes place in sentry (which although hardcoded, will always return a valid entity name). Do you think validation here is necessary? Since entity names are a snuba concept anyways and we aren't directly exposing these names outside (via some API), could we maybe even validate on the API side?

Reply:
Yeah good point. I'll remove it.